rfcs/text/0030/process.yml

```diff
@@ -0,0 +1,158 @@
+        Length of the process.previous_args array.
+
+        This field can be useful for querying or performing bucket analysis on
+        how many arguments were provided to start a process.
+        More arguments may be an indication of suspicious activity.
+      example: 4
+
+    - name: previous_executable
+      level: extended
+      type: keyword
+      description: >
+        Absolute path to the process previous executable.
+      example: /usr/bin/ssh
+      multi_fields:
+        - type: match_only_text
+          name: text
+
+    - name: interactive
+      level: extended
+      type: boolean
+      example: true
+      short: Whether the process is connected to an interactive shell.
+      description: >
+        Whether the process is connected to an interactive shell.
+
+        Process interactivity is infered from the processes file descriptors. If the character device for the controlling tty is the same as stdin and stderr for the process, the process is considered interactive.
+
+        Note: A non-interactive process can belong to an interactive session and is simply one that does not have open file descriptors reading the controlling TTY on FD 0 (stdin) or writing to the controlling TTY on FD 2 (stderr). A backgrounded process is still considered interactive if stdin and stderr are connected to the controlling TTY.
+
+    - name: fds
+      level: extended
+      type: object
+      short: An array of the lowest file descriptors for the process.
+      description: >
+        An array of the lowest file descriptors for the process.
+
+        Typically includes fd0(stdin), fd1(stdout), fd2(stderr), but can include other non character devices such as files, sockets, pipes and FIFOs.
+
+      example: >
+        [
+          {
+            descriptor:0,
+            type: 'char_device',
+            char_device: {
+              major: 8,
+              minor: 1
+            }
+          },
+          {
+            descriptor:1,
+            type:'char_device',
+            char_device: {
+              major: 8,
+              minor: 1
+            }
+          },
+          {
+            descriptor:9,
+            type:'pipe',
+            pipe: {
+              inode: '6183207'
+            }
+          }
+        ]
+
+      normalize:
+        - array
+
+    - name: env_vars
+      level: extended
+      type: object
+      short: Environment variables set at the time of the event.
+      description: >
+        Environment variables set at the time of the event.
+
+        May be filtered to protect sensitive information.
+      example: >
+        {
+          "USER": "elastic",
+          "LANG": "en_US.UTF-8",
+          "HOME": "/home/elastic"
+        }
+
+    - name: entry_meta.type
+      level: extended
+      type: keyword
+      short: The entry mechanism.
+      description: >
+        The entry mechanism.
+
+        Values include: init(e.g systemd), sshd, ssm, kubelet, teleport, terminal, console
```

Comment:
`kubelet` might be better as `container`, that's what we called it in Cmd, and is more accurate since you can run containers without k8s. Though in the future it might be valuable to have a distinction between containers in kubernetes and containers run elsewhere.

Reply:
in the same way that teleport uses the ssh protocol but is named differently than the ssh entry point, I think it makes sense to keep "kublet" as the specific entry point. Access through the kublet tells you a lot (ie it is remotely accessible through kube API) vs container - other things including purely local things like systemd use/create containers.  Also not all things done via or by the kublet necessarily involve containers. 

There may be response cases that benefit from this specificity too - e.g. a vuln in kublet, find all hosts in my network that have had kublet entry service related events in the last month.